### PR TITLE
update the package version and CHANGELOG

### PR DESCRIPTION
**Pull request by Hmone23**
@biplobice 

> https://github.com/concrete5cojp/c5j_ratings/issues/2

I fixed above issue.
Please confirm PR.

### CODE REVIEW

**Comment by biplobice** [comment]
@Hmone23 
1. It’s better to have the page path `/dashboard/rating_history` instead of `/dashboard/ratings_history`. I prefer `/dashboard/ratings`.
2. I think you know about `$autoSortColumns`. If not, please check-
- https://github.com/concrete5/concrete5/blob/85e56adc1516381b3d9cfdd66fc554fabac5b1a6/concrete/src/User/Point/EntryList.php#L12
- https://legacy-documentation.concrete5.org/developers/system/searching-sorting-paging
- https://notnil-creative.com/blog/archives/2565
3. Please follow concrete CMS conventions about naming. `src/Export/ExportRatings.php`
- https://github.com/concrete5/concrete5/blob/f49c756156caaa598749469e5727d0b2bf119a9f/concrete/src/Csv/Export/UserExporter.php
- https://github.com/concrete5/concrete5/blob/9d2f9432cb4746caa1a7ae45f4871a5efbf6b6ac/concrete/src/Express/Export/EntryList/CsvWriter.php

**Comment by Hmone23** [comment]
@biplobice 

Fixed
Please confirm

**Comment by biplobice** [comment]
It would be great if you can make the filtering UI like this-
Screen Shot 2022-02-22 at 11 50 50 AM

File: single_pages/dashboard/ratings/view.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                <tbody>
+                <?php
+                    foreach ($ratings as $rating) {
+                        $page = \Concrete\Core\Page\Page::getByID($rating->getCID());
+                        $ui = $app->make(UserInfoRepository::class)->getByID($rating->getUID());
+                        if($ui){
+                            $name = $ui->getUserName();
+                        }else{
+                            $name = 'unknown';
+                        }
```
Comment (biplobice) [inline single_pages/dashboard/ratings/view.php:53]: - Make the text `unknown` translatable & Capitalize.
- Use proper spacing in your code. You may use `c5:phpcs` command. Or set it to your IDE https://biplobice.github.io/developers-mac/#/?id=install-php-cs-fixer

File: controllers/single_page/dashboard/ratings.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        $this->set('query', $query);
+
+        $pagination = $this->ratingsList->getPagination();
+        $results = $pagination->getCurrentPageResults();
+
+        if ($results) {
+            $this->set('ratingsList', $this->ratingsList);
+            $this->set('ratings', $results);
+            $this->set('pagination', $pagination);
+        }
```
Comment (biplobice) [inline controllers/single_page/dashboard/ratings.php:55]: IMO, the `if ($results)` check is unnecessary. You could already set the variables.

File: controllers/single_page/dashboard/ratings.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    public function csv_export($token = '')
+    {
+        if (!$this->token->validate('export', $token)) {
+            $this->error->add($this->token->getErrorMessage());
+        }
+
+        if (!$this->error->has()) {
+            $headers = [
+                'Content-Type' => 'text/csv',
+                'Content-Disposition' => 'attachment; filename=ratings_history_' . Carbon::now() . '.csv',
```
Comment (biplobice) [inline controllers/single_page/dashboard/ratings.php:75]: As you have changed the page name. IMO, it's better to change the CSV filename as well.

File: controllers/single_page/dashboard/ratings.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    {
+        $r = Request::getInstance();
+        if ($r->query->has('rated_date') && $r->query->get('rated_date') !== '') {
+            $this->ratingsList->filterByRatedDate($r->query->get('rated_date'));
+            $this->set('rated_date', $r->query->get('rated_date'));
+
+        }
+
+        return $this->ratingsList;
+    }
```
Comment (biplobice) [inline controllers/single_page/dashboard/ratings.php:112]: Wrong code design.
By the name `getFilterByRatedDate` I assume you're trying to get a filter object which isn't.
You may try this approach https://github.com/concrete5/concrete5/blob/8.5.x/concrete/controllers/single_page/dashboard/reports/logs.php#L161-L194

File: src/Export/ExportRatings.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+use C5jRatings\Entity\C5jRating;
+use Concrete\Core\Support\Facade\Application;
+use Concrete\Core\User\UserInfoRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Iterator;
+use League\Csv\Writer;
+use C5jRatings\Search\RatingList;
+
+class ExportRatings
```
Comment (biplobice) [inline src/Export/ExportRatings.php:13]: It would be great if you can rename the class to `RatingExporter` and follow this coding https://github.com/concrete5/concrete5/blob/8.5.x/concrete/src/Csv/Export/UserExporter.php

BTW, if you don't know how to refactor a class using PhpStorm https://www.jetbrains.com/help/phpstorm/refactoring-source-code.html

**Review by biplobice — CHANGES_REQUESTED** [review]
@Hmone23 
Your code is working fine but I've added some comments to standardize your code.
It would be great if you can adopt these and apply them in the future as well.
PhpStorm has lots of built-in features and plugins. Try to use & satisfy them.
If you need any help you may check https://biplobice.github.io/developers-mac/ or ask us.

**Comment by Hmone23** [comment]
@biplobice 

Thank for your comments.
Fixed

File: controllers/single_page/dashboard/ratings.php (hunk 4 above)
Comment (biplobice) [reply controllers/single_page/dashboard/ratings.php]: `getFilterByRatedDate ` makes me feel like you're trying to get a filter object. 
If I'm not wrong the reference I sent you has `getFilteredList` which clearly states that you're trying to get a list object which is filtered.

File: single_pages/dashboard/ratings/view.php [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+if(isset($ratingsList, $ratings) && count($ratings) > 0){
+?>
+
+    <div class="table-responsive">
+            <table class="ccm-search-results-table">
+                <thead>
+                    <tr>
+                        <th class="<?=$ratingsList->getSortClassName('r.cID')?>"><a href="<?=$ratingsList->getSortURL('r.cID')?>">Page Name</a></th>
+                        <th class="<?=$ratingsList->getSortClassName('r.uID')?>"><a href="<?=$ratingsList->getSortURL('r.uID')?>">User</a></th>
+                        <th class="<?=$ratingsList->getSortClassName('r.ratedAt')?>"><a href="<?=$ratingsList->getSortURL('r.ratedAt')?>">ratedAt</a></th>
```
[the remark this answers is not in the document]
Comment (biplobice) [reply single_pages/dashboard/ratings/view.php]: Should be `Rated At` instead of `ratedAt`

File: single_pages/dashboard/ratings/view.php (hunk 1 above)
Comment (biplobice) [reply single_pages/dashboard/ratings/view.php]: > Use proper spacing in your code.

Haven't done yet.

File: single_pages/dashboard/ratings/view.php [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+                        $page = \Concrete\Core\Page\Page::getByID($rating->getCID());
+                        $ui = $app->make(UserInfoRepository::class)->getByID($rating->getUID());
+                        if($ui){
+                            $name = $ui->getUserName();
+                        }else{
+                            $name = 'unknown';
+                        }
+                    ?>
+                        <tr>
+                            <td><?=h($page->getCollectionName())?></td>
```
[the remark this answers is not in the document]
Comment (biplobice) [reply single_pages/dashboard/ratings/view.php]: Not fixed yet.

File: src/Export/ExportRatings.php (hunk 5 above)
Comment (biplobice) [reply src/Export/ExportRatings.php]: Not done yet.
If you don't want to change the class name you should tell us. Perhaps, explain why.

File: src/Export/ExportRatings.php [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+     * @return array
+     */
+    private function projectEntry(C5jRating $rating): Iterator
+    {
+        $page = \Concrete\Core\Page\Page::getByID($rating->getCID());
+        $ui = $this->app->make(UserInfoRepository::class)->getByID($rating->getUID());
+        if($ui){
+            $name = $ui->getUserName();
+        }else{
+            $name = 'unknown';
```
Comment (biplobice) [inline src/Export/ExportRatings.php:87]: You should keep the text similar.

File: single_pages/dashboard/ratings/view.php [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+                </thead>
+                <tbody>
+                <?php
+                foreach ($ratings as $rating) {
+                    $page = \Concrete\Core\Page\Page::getByID($rating->getCID());
+                    $ui = $app->make(UserInfoRepository::class)->getByID($rating->getUID());
+                    if($ui){
+                        $name = h($ui->getUserName());
+                    }else{
+                        $name = t('Unknown');
```
Comment (Hmone23) [inline single_pages/dashboard/ratings/view.php:62]: @biplobice 

I have fixed it.

File: single_pages/dashboard/ratings/view.php (hunk 1 above)
Comment (Hmone23) [reply single_pages/dashboard/ratings/view.php]: > > Use proper spacing in your code.
> 
> Haven't done yet.

I have fixed it.
Please confirm

File: controllers/single_page/dashboard/ratings.php (hunk 4 above)
Comment (Hmone23) [reply controllers/single_page/dashboard/ratings.php]: > `getFilterByRatedDate ` makes me feel like you're trying to get a filter object.
> If I'm not wrong the reference I sent you has `getFilteredList` which clearly states that you're trying to get a list object which is filtered.

Yes, I fixed it according your reference code.
https://github.com/concrete5cojp/c5j_ratings/blob/feature/Dashboard_Report/controllers/single_page/dashboard/ratings.php#L78-L90

File: src/Export/ExportRatings.php (hunk 5 above)
Comment (Hmone23) [reply src/Export/ExportRatings.php]: > Not done yet.
> If you don't want to change the class name you should tell us. Perhaps, explain why.

Sorry, I forgot to commit this file.

File: src/Export/ExportRatings.php (hunk 8 above)
Comment (Hmone23) [reply src/Export/ExportRatings.php]: > You should keep the text similar.

I fixed it.
Please confirm.

File: controllers/single_page/dashboard/ratings.php (hunk 4 above)
Comment (biplobice) [reply controllers/single_page/dashboard/ratings.php]: I wanted you to change the method name to `getFilteredList` as you're getting the list object not the filter object.

File: single_pages/dashboard/ratings/view.php (hunk 1 above)
Comment (biplobice) [reply single_pages/dashboard/ratings/view.php]: Sorry, I think I couldn't explain you.
Please check this documentation https://www.php-fig.org/psr/psr-2/
You don't use proper spacing on an if condition.
You can fix those at once by using the `c5:phpcs` command or setting external tools to the PhpStorm. See https://cs.symfony.com/

File: controllers/single_page/dashboard/ratings.php (hunk 4 above)
Comment (Hmone23) [reply controllers/single_page/dashboard/ratings.php]: > I wanted you to change the method name to `getFilteredList` as you're getting the list object not the filter object.

Sorry, I fixed it.

File: single_pages/dashboard/ratings/view.php (hunk 1 above)
Comment (Hmone23) [reply single_pages/dashboard/ratings/view.php]: > Sorry, I think I couldn't explain you.
> Please check this documentation https://www.php-fig.org/psr/psr-2/
> You don't use proper spacing on an if condition.
> You can fix those at once by using the `c5:phpcs` command or setting external tools to the PhpStorm. See https://cs.symfony.com/

Yes, I ran it using ` concrete/bin/concrete5 c5:phpcs fix packages/c5j_ratings`
but it didn't change
Screen Shot 2022-02-25 at 12 38 29 PM

Comment (biplobice) [reply single_pages/dashboard/ratings/view.php]: https://www.php-fig.org/psr/psr-2/#51-if-elseif-else
This is how to write an if-else condition.
The `phpcs` command should fix these. If it doesn't you may want to make an issue on `concrete5` repo.
Anyway, I'm gonna fix this for you.

**Comment by biplobice** [comment]
@Hmone23 
It looks much better now.
I would like to request you a couple of more improvements.

1. The search box and the table has different date format. Please use the same.
   Hint: Check core, don't use any custom format as the format will be different based on language or timezone.
2. Search result includes results other than the specified one.

Screen Shot 2022-02-25 at 5 54 48 PM

**Comment by Hmone23** [comment]
@biplobice 

> Search result includes results other than the specified one.

 Search result includes results other than the specified one because `createdAt` field is date time format.
it will search from the searched date.

```
 public function filterByRatedDate($date, $comparison = '>=')
    {
        $this->query->andWhere($this->query->expr()->GTE('r.ratedAt', ':date'));
        $this->query->setParameter('date', $date);
    }
```
Screen Shot 2022-02-25 at 6 28 23 PM

**Comment by Hmone23** [comment]
@biplobice 

> The search box and the table has different date format. Please use the same.
> Hint: Check core, don't use any custom format as the format will be different based on language or timezone.

thank for hint. Fixed

**Comment by biplobice** [comment]
> it will search from the searched date.

Well, you should use `Rated After` (exclusive, ratedAt > ratedAfter) or `Rated From` (inclusive, , ratedAt >= ratedFrom) then.

Also, FYI, core uses placeholder instead of a label.

**TLDR; Remove `Rated Date` label from on top of the search box and add the placeholder `Rated From`.**

**Comment by Hmone23** [comment]
@biplobice 

> Also, FYI, core uses placeholder instead of a label.
> TLDR; Remove Rated Date label from on top of the search box and add the placeholder Rated From.

I don't know how to use placeholders in the core date picker.
Please tell me how to fix.

**Comment by biplobice** [comment]
Hmm, there is no such an option to add a placeholder in core.
You may add a piece of javascript code-
```
<script>
    $('#rated_date_pub').attr('Placeholder', "<?= t('Rated From') ?>");
</script>
```

**Comment by Hmone23** [comment]
@biplobice 

Thanks
Fixed

**Comment by biplobice** [comment]
@Hmone23 
Oops! Sorry!
Please update the changelog.
It should be `### Added` not `### Fixed`
